python/pyspark/sql/tests/arrow/test_arrow_udtf.py

```diff
@@ -382,6 +382,82 @@ def analyze(arg):
 
         self.assertIn("INVALID_UDTF_BOTH_RETURN_TYPE_AND_ANALYZE", str(cm.exception))
 
+    def test_arrow_udtf_with_table_argument_basic(self):
+        @arrow_udtf(returnType="filtered_id bigint")  # Use bigint to match int64
+        class TableArgUDTF:
+            def eval(self, table_data: "pa.RecordBatch") -> Iterator["pa.Table"]:
+                assert isinstance(
+                    table_data, pa.RecordBatch
+                ), f"Expected pa.RecordBatch, got {type(table_data)}"
+
+                # Convert record batch to table to work with it more easily
+                table = pa.table(table_data)
+
+                # Filter rows where id > 5
+                id_column = table.column("id")
+                mask = pa.compute.greater(id_column, pa.scalar(5))
+                filtered_table = table.filter(mask)
+
+                if filtered_table.num_rows > 0:
+                    result_table = pa.table(
+                        {"filtered_id": filtered_table.column("id")}  # Keep original type (int64)
+                    )
+                    yield result_table
+
+        # # Test with DataFrame API using asTable()
+        # input_df = self.spark.range(8)
+        # result_df = TableArgUDTF(input_df.asTable())
+        expected_df = self.spark.createDataFrame([(6,), (7,)], "filtered_id bigint")
+        # assertDataFrameEqual(result_df, expected_df)
```

Comment:
Is this not working? If not working now and will be fixed later, shall we file a ticket and leave a TODO comment?